grug/math/src/test_utils.rs

```diff
@@ -189,3 +192,226 @@ macro_rules! int_test {
+                fn [<$name _udec128>]() {
+                    ($test_fn)(<$crate::Udec128 as $crate::NumberConst>::ZERO, $($pu128),*);
+                }
+            )?
+
+            $(
+                #[test]
+                #[allow(clippy::just_underscores_and_digits)]
+                fn [<$name _udec256>]() {
+                    ($test_fn)(<$crate::Udec256 as $crate::NumberConst>::ZERO, $($pu256),*);
+                }
+            )?
+
+            $(
+                #[test]
+                #[allow(clippy::just_underscores_and_digits)]
+                fn [<$name _dec128>]() {
+                    ($test_fn)(<$crate::Dec128 as $crate::NumberConst>::ZERO, $($pi128),*);
+                }
+            )?
+
+            $(
+                #[test]
+                #[allow(clippy::just_underscores_and_digits)]
+                fn [<$name _dec256>]() {
+                    ($test_fn)(<$crate::Dec256 as $crate::NumberConst>::ZERO, $($pi256),*);
+                }
+            )?
+        }
+    };
+    // Multiple optional passing/failing tests.
+    (
+        $name:ident
+        inputs = {
+            $(udec128 = {
+                passing: [$($pu128:expr),* $(,)?] $(,)?
+                $(failing: [$($fu128:expr),* $(,)?])? $(,)?
+            } $(,)? )?
+            $(udec256 = {
+                passing: [$($pu256:expr),* $(,)?] $(,)?
+                $(failing: [$($fu256:expr),* $(,)?])? $(,)?
+            } $(,)? )?
+            $(dec128 = {
+                passing: [$($pi128:expr),* $(,)?] $(,)?
+                $(failing: [$($fi128:expr),* $(,)?])? $(,)?
+            } $(,)? )?
+            $(dec256 = {
+                passing: [$($pi256:expr),* $(,)?] $(,)?
+                $(failing: [$($fi256:expr),* $(,)?])? $(,)?
+            } $(,)? )?
+        } $(,)?
+        $(attrs = $ (#[$meta:meta])*)? $(,)?
+        method = $test_fn:expr
+    ) => {
+        dec_test!(
+            $name
+            inputs = {
+                $(udec128 = [[$($pu128),*] $(, [$($fu128),*])?])?
+                $(udec256 = [[$($pu256),*] $(, [$($fu256),*])?])?
+                $(dec128 = [[$($pi128),*] $(, [$($fi128),*])?])?
+                $(dec256 = [[$($pi256),*] $(, [$($fi256),*])?])?
+            }
+            $(attrs = $(#[$meta])*)?
+            method = $test_fn
+        );
+    };
+}
+
+pub fn int<U>(val: &str) -> Int<U>
+where
+    Int<U>: FromStr,
+    <Int<U> as FromStr>::Err: Debug,
+{
+    Int::from_str(val).unwrap()
+}
+
+/// Shortcut for create a `Dec` from a string.
+pub fn dec<U>(val: &str) -> Dec<U>
+where
+    Dec<U>: FromStr,
+    <Dec<U> as FromStr>::Err: Debug,
+{
+    Dec::from_str(val).unwrap()
+}
+
+pub trait Leftover {
+    /// Rappresent the minimum value that can be represented by the decimal.
+    ///
+    /// `Dec::raw(Int::<U>::ONE)`
+    const LEFTOVER: Self;
+}
+
+impl<U> Leftover for Dec<U>
+where
+    Int<U>: NumberConst,
+{
+    const LEFTOVER: Dec<U> = Dec::raw(Int::<U>::ONE);
+}
```

Comment:
I think this can be added to the `FixedPoint` trait?

Also "left over" isn't a good name. Maybe "step" or "tick"?

Reply:
I've added in test utils because idk if this has some needed outside test purpose